src/components/form/Form.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useState} from "react";
+import classNames from "classnames";
+
+import {ApiErrorShape} from "../../core/network-manager/networkModels";
+import {generateErrorMessage} from "../../utils/error/errorUtils";
+import {ArrayToUnion} from "../../utils/typeUtils";
+import {ERROR_TYPES} from "../../utils/error/errorConstants";
+
+export interface FormProps {
+  testid: string;
+  onSubmit: () => void;
+  children: React.ReactNode;
+  errorInfo: ApiErrorShape | null;
+  knownErrorKeys: string[] | null;
+  onChange?: (event: React.SyntheticEvent) => void;
+  customClassName?: string;
+  isSubmitAllowed?: boolean;
+  shouldDisplayFieldErrorCountInErrorMessage?: boolean;
+  skipErrorTypes?: Array<ArrayToUnion<typeof ERROR_TYPES>>;
+}
+
+function Form(props: FormProps) {
+  const {
+    children,
+    testid,
+    onSubmit,
+    customClassName,
+    errorInfo,
+    knownErrorKeys,
+    isSubmitAllowed = true,
+    skipErrorTypes
+  } = props;
+  const [isPristine, setPristine] = useState(true);
+  const containerClassName = classNames("form", customClassName);
+  const formError = generateErrorMessage(errorInfo, {
+    knownErrorKeys,
+    skipTypes: skipErrorTypes
+  });
+
+  return (
+    <form
+      className={containerClassName}
+      data-testid={testid}
+      onChange={handleFormChange}
+      onSubmit={handleSubmit}>
+      {Boolean(formError) && (
+        <p data-testid={`${testid}.form-error-message`}>{formError}</p>
```

Comment:
Let's add a class name to the error message so we can customize the style easily 
```suggestion
        <p className={"form-error-message"} data-testid={`${testid}.form-error-message`}>{formError}</p>
```

Reply:
Done! I also added basic styles.